### ISSUE TITLE
An overview of learning to rank for information retrieval.

### ISSUE THREAD

**Issue by sooooner** [sooooner/papers#56]
# An overview of learning to rank for information retrieval. [[link](https://ieeexplore.ieee.org/document/5170911)]

**Abstract**
Learning to rank에 대한 overview

## Introduction 
IR 모델은 document representation, user query representation, ranking of retrieved documents 등을 정보 검색 프로세스의 주요 단계를 정의하고 설명하는 수학적 프레임 워크
기존 IR model은 4가지 종류가 있음
1. Boolean model
2. Vector space model
3. Probabilistic model
4. Language model

기존 모델의 단점은 1. 파라미터가 휴리스틱 2. 정확도 낮음 -> LTR은 이런 단점 극복함

LTR은 2가지 단계로 진행됨
1. Learning P(Rnak | q, theta)
2. scoring -> ranking

ranking에 ML을 쓴는 이유는
1. Clf, Reg와 본질을 공유함 (ex. -Q - D간의 서로 다른 관련성을 할당하는 것)
2. 기존 방법 보다 성능이 좋음

LTR는 이전 구매 내역을 기반으로 한 제품 추천 ranking, 웹 페이지 콘텐츠에 대한 광고 ranking, 뉴스기사 ranking등등 사용자의 정보 요구 사항을 제공하는 모든 측면에서 중요한 작업임.

## Related Concepts 
**Definition 1:  A ranking is defined as an ordering on the set of alternatives**
ranking을 매기기 위해 두가지 작업이 필요함
1. document collection : alternatives set을 얻는것 
2. ordering

**Definition 2: Learning to Rank is to build ranking function F on training samples and to rank samples if and only if the testing performance satisfies with some requirement R which is measured by some performance P.**
: ranking을 지정하는 함수 F를 빌드하고, 성능 P로 측정 가능한 요구조건 R을 충족하는 ranking 지정

![image](https://user-images.githubusercontent.com/31437443/124208366-5bc5e900-db22-11eb-9ab5-bf64261af33e.png)

## Learning to Rank Models 
LTR모델은 1. point wise, 2. pairwise, 3. listwise 모델로 분류됨(training data는 각각 개별 문서, 문서 쌍, 문서 목록임)

### Pointwise Models 
개별 문서로 훈련되는 모델 : 관련 쿼리에 개별 문서들이 점수할당됨. 출력은 문서들의 내림차순 목록

Ramesh는 ranking model을 빌드 하기 위해서 Maximum Entropy regression model을 적용함, weight를 얻기위해 GD사용

![image](https://user-images.githubusercontent.com/31437443/124209155-fa067e80-db23-11eb-81a0-a9c3aa4ac26d.png)

그림에서 training document는 모델에 의해 관련, 무관 문서로 분류될수 있음. 또한 document feature를 얻음 
장점 : feature weight를 자동으로 조절함
단점 : unseen data에서 별로임, 서로 다른 문서지만 feature가 같으면 같은 점수 할당

---

Ping Li는 classification idea를 적용하여 class 관련성 확률을 ranking score로 변환하는 McRank model 구성함.
class 관련성 확률은 샘플이 다른 클래스에 속하는 정도를 측정하는 것.
분류의 효과가 좋을수록 DCG가 높아짐. 
ME 모델과 차이점은 학습 절차임. -> 그림으로 확인

![image](https://user-images.githubusercontent.com/31437443/124209824-38506d80-db25-11eb-8bff-c14f7f1787d8.png)

장점 : 분류 효과가 좋을수록 DCG 좋음
단점 : black box, 특정 데이터에서만 성능이 좋음, relevance level을 보여주지 않음

--- 

이 외에 회귀, 분류 모델이 point wise ranking model에 적용됨.
Caruana는 NN을 기반으로한 RankProp제안 : 점수를 빠르게 할당하지만 훈련시간을 많이 소비함
Joachims는 SVM을 사용 : 정밀도는 우수하지만 일부 관련 없는 문서가 관련된다고 할때가 있음

---

point wise model의 공통적인 두가지 장점
1. 분류와 회귀 방법이 실제로 ranking 정밀도를 향상 시킬수 있음
2. 훈련 데이터에서 학습을 통해 파라미터가 조정됨.
단점
1. training D가 하나씩 개별적으로 사용되기 때문에 D간 종속성이 고려되지 않음 
-> 이런 단점은 pairwise model로 부분적으로 해결 할 수 있음

### Pairwise Models 
Pairwise 모델은 사람에 의해 점수가 매겨진 문서쌍으로 학습됨.

Burges는 NN을 기반으로 한 RankNet 모델을 제시함
RankProp과 차이점은 다른 문서쌍으로 훈련되고 예상 ranking 결과와 모델에 의해 생성된 ranking 결과 간의 차이를 나타내는 loss function임.

training 절차는 생성된 ranking과 기본 ranking간의 차이를 최소화 하는것 -> cross entropy theory를 기반으로 함

![image](https://user-images.githubusercontent.com/31437443/124211500-27552b80-db28-11eb-84ef-125c4ff7d2d6.png)

loss function에서 pij는 문서 i가 문서 j 보다 순위가 높을 확률, 사람들이 설정한 Pij tilde는 문서 i가 문서 j보다 높은 순위에 있도록 원하는 근사 확률.  point wise model과 차이점은 아래 그림에서 볼수 있음

![image](https://user-images.githubusercontent.com/31437443/124211541-3dfb8280-db28-11eb-8420-e3738a8740de.png)

장점
1. 주어진 학습 문서 쌍에 대해 최종 순위를 변경하지 않고 출력에 임의의 오프셋을 삽입 할 수 있다는 것입니다. 
2. RankProp보다 훈련시간이 짧음(절반)
단점
1. 학습 알고리즘은 loss에 대한 정규화 된 분석을 위한 손실의 상한이 없기 때문에 완벽하지 않음

---

Yoav는 AdaBoost에서 영감을 얻어 RankBoost 모델 제안. 이는 **"weak rankers"**를 결합하여 weak ranker의 가중 합계 인 더 정확한 ranker를 형성하는 것. 학습 절차에서는 최적의 ranker를 구성하기 위해 약한 랭커의 가중치를 최적화함
장점 
1. 중요한 feature에 따라 구성된 weak ranker에 무거운 가중치가 할당되기 때문에 중요한 문서 feature가 최종 순위에 훨씬 더 많은 영향을 미칠 수 있음
2. regression ranking, Nearest Neighbor ranking, and Vector Similarity Ranking보다 성능이 좋음
단점
1. weak ranker를 결합하여 강력한 ranker를 형성 할 수 있는 최적의 방법이 없음

---

pair wise model의 공통적인 장점
1. 훈련 문서가 쌍이기 때문에 문서 독립성이 깨짐
단점
1. 문서 쌍에서 종속성만 고려됨 -> 전체 순위에서의 종속성을 고려 할 수 없음.

### Listwise Models
문서 목록으로 훈련 되어 ranking을 매기는 모델. 각 list에서 문서는 쿼리에 대해 서로 다른 관련성 점수로 지정됨.

Jun Xu와 Hang Li는 AdaBoost에서 영감을 얻은 AdaRank 모델 제안. 학습 절차는 아래와 같습니다.

![image](https://user-images.githubusercontent.com/31437443/124215015-32ab5580-db2e-11eb-9175-78bdbe80f433.png)

장점 
1.  pair, point wise model보다 더 나은 문서 종속성을 구축함. 
단점 
1. flexibility는 사람들이 예상하는 정밀도를 보장할 만큼 충분히 강하지 않음

### Difficulties 
Learning to Rank에도 분류, 회귀가 직면한 어려움이 존재함. 
1. 모델이 지속적인 학습을 할 수 없음!
2. 학습을 위한 충분한 train set을 구성하는 것이 힘듦

## Estimation Measures
ranking result를 평가하기 위한 측도는  Normalized Discount Cumulative Gain (NDCG), Mean Average Precision (MAP)가 있다.
